include/mapped_soa.h

```diff
@@ -85,6 +85,27 @@ class MappedSoA {
     key_to_index_.erase(key);
   }
 
+  void copy(size_t index0, size_t index1) {
+    const size_t size_val = size();
+    if (size_val <= index0 || size_val <= index1) {
+      assert(false);
+
+      return;
+    }
+
+    if (index0 == index1) {
```

Comment:
isn't this just a protection to skip copying the same data over itself?  I'm going to reuse your example here:
```C++
// Map of String Key type to a position.
MappedSoA<std::string, Vector3> positions;
positions["Orc1"] = Vector::Zero;
positions.Copy("Orc1", "Orc1");  // skipped because the two indices are the same
```

Reply:
Yes, but the soa_.copy is going to do it for us anyway :)

Reply:
oh! right! :dizzy: